lib/rspec/que/queue_up.rb

```diff
@@ -1,89 +1,155 @@
 require 'rspec/mocks/argument_list_matcher'
+require 'time'
 
 module RSpec
   module Que
     module Matchers
       class QueueUp
         def initialize(job_class = nil)
+          @matchers = [QueuedSomething.new]
+          @matchers << QueuedClass.new(job_class) if job_class
           @job_class = job_class
+
+          @stages = nil
         end
 
         def matches?(block)
           @before_jobs = enqueued_jobs.dup
           block.call
-          enqueued_something? && enqueued_correct_class? && with_correct_args?
+
+          @stages = []
```

Comment:
You may consider only saving the `first_failed_stage` and use it in `failure_message` instead of doing the search then.